eq_translations/validate_translation.py

```diff
@@ -0,0 +1,76 @@
+from jsonpointer import resolve_pointer
+
+from eq_translations.utils import (
+    list_pointers,
+    find_pointers_to,
+)
+
+
+def compare_schemas(source_schema, target_schema):
+    """
+    Compare the pointers in two json structures and return differences
+    :param source_schema: Structure to identify differences against
+    :param target_schema: Target structure to compare against
+    :return:
+    """
+    source_survey_pointers = set(list_pointers(source_schema))
+    target_survey_pointers = set(list_pointers(target_schema))
+
+    missing_target_pointers = source_survey_pointers.difference(target_survey_pointers)
+    missing_source_pointers = target_survey_pointers.difference(source_survey_pointers)
+
+    missing_non_plural_target_pointers = {
+        pointer for pointer in missing_target_pointers if "text_plural" not in pointer
+    }
+    missing_non_plural_source_pointers = {
+        pointer for pointer in missing_source_pointers if "text_plural" not in pointer
+    }
+
+    missing_pointers = (
+        missing_non_plural_target_pointers | missing_non_plural_source_pointers
+    )
+
+    if missing_pointers:
+        print(
+            f"\nDifferences between source/target schema strings (excluding text_plural): {len(missing_pointers)}"
+        )
+        print(f"Total strings in source schema: {len(source_survey_pointers)}")
+        print(f"Total strings in target schema: {len(target_survey_pointers)}\n")
+
+        if missing_target_pointers:
+            print(
+                "Pointers in the source schema that are missing from the target schema:"
+            )
+            for pointer in missing_target_pointers:
+                print(pointer)
+
+        if missing_source_pointers:
+            print(
+                "Pointers in the target schema that are missing from the source schema:"
+            )
+            for pointer in missing_source_pointers:
+                print(pointer)
+
+    return missing_pointers
```

Comment:
- There is no output if there are no differences, so it's not clear if the script did anything.
- When there are differences, the plural differences are also listed. For example I removed a property from the Welsh test_language and got:
```
Differences between source/target schema strings (excluding text_plural): 1
Total strings in source schema: 144
Total strings in target schema: 151

Pointers in the source schema that are missing from the target schema:
/sections/0/groups/0/blocks/0/question/title
Pointers in the target schema that are missing from the source schema:
/sections/0/groups/0/blocks/3/question/title/text_plural/forms/two
/sections/0/groups/0/blocks/3/question/answers/0/options/0/label/text_plural/forms/two
/sections/0/groups/0/blocks/3/question/title/text_plural/forms/few
/sections/0/groups/0/blocks/3/question/answers/0/options/0/label/text_plural/forms/few
/sections/0/groups/0/blocks/3/question/answers/0/options/0/label/text_plural/forms/zero
/sections/0/groups/0/blocks/3/question/title/text_plural/forms/zero
/sections/0/groups/0/blocks/3/question/answers/0/options/0/label/text_plural/forms/many
/sections/0/groups/0/blocks/3/question/title/text_plural/forms/many
```

Reply:
- Now outputs feedback when there are no missing pointers.
- As for the second issue, it was because I was using the wrong key, which is now fixed.